sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py

```diff
@@ -124,7 +124,8 @@ async def open(self):
             self.session = aiohttp.ClientSession(**clientsession_kwargs)
         # pyright has trouble to understand that self.session is not None, since we raised at worst in the init
         self.session = cast(aiohttp.ClientSession, self.session)
-        await self.session.__aenter__()
+        if self.session is not None:
```

Comment:
for the issue that this is fixing, shouldn't we raise here else the user will get a `NoneType` error when they try and do some other operation with the transport?

I am puzzled as to how session became None at this point?

Reply:
had an OOB discussion with xiang that this is a revert back to the original code.